File: c++/S01-elemental-actions/E02-quadratic-roots.cpp
#include <iostream>
#include <cmath>

int main() {
	float quadratic, lineal, independent;

	std::cout << "\n\e[0;35m[========= QUADRATIC ROOTS =========]\e[0m\n" << '\n';

	std::cout << "Enter the quadratic coefficient: ";
	std::cin >> quadratic;
	std::cout << "Enter the lineal coefficient: ";
	std::cin >> lineal;
	std::cout << "Enter the independent term: ";
	std::cin >> independent;

	float discriminant = lineal * lineal - 4 * quadratic * independent;

	float firstRoot = (-1 * lineal + sqrt(discriminant)) / (2 * quadratic);
	float secondRoot = (-1 * lineal - sqrt(discriminant)) / (2 * quadratic);

	std::cout << "\nThe first root is: " << firstRoot << '\n';
	std::cout << "The second root is: " << secondRoot << '\n';

	return 0;
}

File: python/S01-elemental-actions/E02-quadratic-roots.py
def main():
	print("\n\033[35m[========= QUADRATIC ROOTS =========]\033[0m\n")

	a: float = float(input("Enter the quadratic value: "))
	b: float = float(input("Enter the lineal value: "))
	c: float = float(input("Enter the independent value: "))

	discriminant: float = b ** 2 - 4 * a * c

	firstRoot: float = (-b + discriminant ** 0.5) / (2 * a)
	secondRoot: float = (-b - discriminant ** 0.5) / (2 * a)

	print(f"\nThe roots of the quadratic equation are: {firstRoot} and {secondRoot}\n")


if __name__ == "__main__":
	main()
